test/integration/utils/PriceManipulation.sol

```diff
@@ -0,0 +1,209 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.22;
+
+import "forge-std/Test.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IV3SwapRouter } from "@uniswap/swap-router-contracts/contracts/interfaces/IV3SwapRouter.sol";
+import { ITakerOracle } from "../../../src/interfaces/ITakerOracle.sol";
+
+library PriceManipulationLib {
+    using SafeERC20 for IERC20;
+
+    // Price movement configuration
+    uint constant STEPS = 5;
+    uint constant STEP_DELAY = 60 seconds;
+    uint constant MAX_ATTEMPTS = 5;
+    uint constant BIPS_BASE = 10_000;
+
+    function movePriceUpPastCallStrike(
+        Vm vm,
+        address swapRouter,
+        address whale,
+        IERC20 cashAsset,
+        IERC20 underlying,
+        ITakerOracle oracle,
+        uint callStrikePercent,
+        uint24 poolFee,
+        uint amountForCallStrike
+    ) external returns (uint finalPrice) {
+        uint currentPrice = oracle.currentPrice();
+        uint targetPrice = (currentPrice * callStrikePercent / BIPS_BASE) + 1;
+
+        finalPrice = moveToTargetPrice(
+            vm,
+            swapRouter,
+            whale,
+            cashAsset,
+            underlying,
+            oracle,
+            targetPrice,
+            true,
+            amountForCallStrike,
+            poolFee
+        );
+        require(finalPrice > targetPrice, "Price did not move past call strike");
+    }
+
+    function movePriceDownPastPutStrike(
+        Vm vm,
+        address swapRouter,
+        address whale,
+        IERC20 cashAsset,
+        IERC20 underlying,
+        ITakerOracle oracle,
+        uint putStrikePercent,
+        uint24 poolFee,
+        uint amountForPutStrike
+    ) external returns (uint finalPrice) {
+        uint currentPrice = oracle.currentPrice();
+        uint targetPrice = (currentPrice * putStrikePercent / BIPS_BASE) - 1;
+
+        finalPrice = moveToTargetPrice(
+            vm,
+            swapRouter,
+            whale,
+            cashAsset,
+            underlying,
+            oracle,
+            targetPrice,
+            false,
+            amountForPutStrike,
+            poolFee
+        );
+        require(finalPrice < targetPrice, "Price did not move past put strike");
+    }
+
+    function movePriceUpPartially(
+        Vm vm,
+        address swapRouter,
+        address whale,
+        IERC20 cashAsset,
+        IERC20 underlying,
+        ITakerOracle oracle,
+        uint24 poolFee,
+        uint amountForPartialMove
+    ) external returns (uint finalPrice) {
+        vm.startPrank(whale);
+        cashAsset.forceApprove(swapRouter, amountForPartialMove);
```

Comment:
reuse `swapCash` here, or remove `movePriceUpPartially` completely and instead use `swapCash` directly where it's used now?